apps/public/src/app/(general)/bijlage/Attachments.tsx

```diff
@@ -0,0 +1,131 @@
+
+export type UploadedFiles = { file: File; id: number }
+
+export const Attachments = ({ meldingId, token }: Props) => {
+  const formRef = useRef<HTMLFormElement>(null)
+  const [uploadedFiles, setUploadedFiles] = useState<UploadedFiles[]>([])
+  const [errorMessage, setErrorMessage] = useState<string>()
+
+  const handleOnChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage(undefined)
+
+    if (!event.currentTarget.files) return
+
+    const files = Array.from(event.currentTarget.files)
+
+    try {
+      if (files.length + uploadedFiles.length > MAX_FILES) {
+        throw new Error(`U kunt maximaal ${MAX_FILES} bestanden uploaden.`)
+      }
+
+      const result = await Promise.all(
+        files.map(async (file) => {
+          const uploadedFile = await postMeldingByMeldingIdAttachment({
+            formData: { file },
+            meldingId,
+            token,
+          })
+
+          return { file, id: uploadedFile.id }
+        }),
+      )
+      setUploadedFiles((currentFiles) => [...currentFiles, ...result])
+    } catch (error) {
+      formRef.current?.reset()
+      setErrorMessage((error as ApiError).message)
+    }
+  }
+
+  const removeFile = async (attachmentId: number) => {
+    setErrorMessage(undefined)
+
+    try {
+      await deleteMeldingByMeldingIdAttachmentByAttachmentId({
+        meldingId,
+        attachmentId,
+        token,
+      })
+      setUploadedFiles((files) => files.filter((file) => file.id !== attachmentId))
+    } catch (error) {
+      setErrorMessage((error as ApiError).message)
+    }
+  }
+
+  return (
+    <Grid paddingBottom="large" paddingTop="medium">
+      <Grid.Cell span={{ narrow: 4, medium: 6, wide: 6 }} start={{ narrow: 1, medium: 2, wide: 3 }}>
+        <BackLink href="/locatie">Vorige vraag</BackLink>
+        <Heading className="ams-mb--sm">Foto’s</Heading>
+        <form ref={formRef} action={redirectToNextPage}>
+          <Field invalid={Boolean(errorMessage)} className="ams-mb--sm">
+            <Label htmlFor="file-upload" optional>
+              Heeft u een foto om toe te voegen?
+            </Label>
+
+            <Column id="file-upload-description" className="ams-mb--sm">
+              <Paragraph>
+                Voeg een foto toe om de situatie te verduidelijken. Verwijder alle persoonsgegevens van u en derden.
+              </Paragraph>
+              <UnorderedList>
+                <UnorderedList.Item>U kunt maximaal 3 bestanden tegelijk toevoegen.</UnorderedList.Item>
+                <UnorderedList.Item>Toegestane bestandtypes: jpg, jpeg en png.</UnorderedList.Item>
+                <UnorderedList.Item>Een bestand mag maximaal 20 MB groot zijn.</UnorderedList.Item>
+              </UnorderedList>
+            </Column>
+
+            {errorMessage && <ErrorMessage id="error-message">{errorMessage}</ErrorMessage>}
+
+            <FileInput handleOnChange={handleOnChange} id="file-upload" hasErrorMessage={Boolean(errorMessage)} />
```

Comment:
Sorry, going back and forth on this a little bit, but: maybe we should set `aria-describedby` here and pass it to the input? That way, all `id`s are defined and used in the same file.

We could even go a bit further and turn it into a generic component that can be moved to `@meldingen/ui`. It already almost is.